File: backend/src/main/java/ch/morgias/cookgenda/models/shopping/ShoppingList.java
package ch.morgias.cookgenda.models.shopping;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
public class ShoppingList {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToMany(
            mappedBy = "shoppingList",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private Set<ShoppingListFood> shoppingListFoods = new HashSet<>();
    private LocalDateTime fromDate;
    private LocalDateTime toDate;
}

File: android/app/src/main/java/ch/morgias/cookgenda/android/models/ShoppingList.kt
package ch.morgias.cookgenda.android.models

import ch.morgias.cookgenda.android.models.utils.LocalDateTimeSerializer
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class ShoppingList(
    val id: Long,
    @Serializable(with = LocalDateTimeSerializer::class) @Contextual val fromDate: LocalDateTime,
    @Serializable(with = LocalDateTimeSerializer::class) @Contextual val toDate: LocalDateTime,
    val shoppingListFoods: Collection<ShoppingListFood>
)

@Serializable
data class ShoppingListWithFoodListByCategory(
    val id: Long,
    @Serializable(with = LocalDateTimeSerializer::class) @Contextual val fromDate: LocalDateTime,
    @Serializable(with = LocalDateTimeSerializer::class) @Contextual val toDate: LocalDateTime,
    val shoppingListFoods: Map<String, MutableList<ShoppingListFood>>
)